### PR TITLE
feat(dropdown-image-and-text): added dropdown image and text

### LINKED ISSUE #2802: Dropdown enhancement - add image and text to dropdown
Add image and text to dropdown

Screen Shot 2020-03-10 at 10 13 04 AM

### PR DESCRIPTION
**Pull request by mattnolting**
closes #2802 
related to #2773

### CODE REVIEW

File: src/patternfly/components/Dropdown/dropdown-group.hbs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<{{#if dropdown-group--type}}{{dropdown-group--type}}{{else}}section{{/if}} class="pf-c-dropdown__group{{#if dropdown-group--modifier}} {{dropdown-group--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown-group.hbs]: can we also update the markup in https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Dropdown/dropdown-menu-groups.hbs to use the hbs partial?

File: src/patternfly/components/Dropdown/dropdown-menu-toggle-image-text.hbs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<div class="pf-c-dropdown__menu{{#if dropdown-menu--modifier}} {{dropdown-menu--modifier}}{{/if}}"
+  aria-labelledby="{{id}}-button"
+  {{#unless dropdown--IsExpanded}}hidden{{/unless}}
+  {{#if dropdown-menu--attribute}}
+    {{{dropdown-menu--attribute}}}
+  {{/if}}>
+
+  {{#> dropdown-group}}
+    {{#> dropdown-menu-item dropdown-menu-item--type="div" dropdown-menu-item--modifier="pf-m-text"}}
+      Foo text
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown-menu-toggle-image-text.hbs]: Can you update this so it's something like "Text" and the second one is "More text", or something like that?

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 3]
```diff
@@ -13,7 +13,7 @@
   --pf-c-dropdown__toggle--BorderWidth: var(--pf-global--BorderWidth--sm);
   --pf-c-dropdown__toggle--BorderTopColor: var(--pf-global--BorderColor--300);
   --pf-c-dropdown__toggle--BorderRightColor: var(--pf-global--BorderColor--300);
-  --pf-c-dropdown__toggle--BorderBottomColor: var(--pf-global--BorderColor--200);
+  --pf-c-dropdown__toggle--BorderBottomColor: var(--pf-global--BorderColor--100);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss]: Was this on purpose? The border color should match other form control elements. Currently it looks like this:

Screen Shot 2020-03-10 at 5 31 22 PM

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 4]
```diff
@@ -81,6 +81,7 @@
   --pf-c-dropdown__menu-item--disabled--Color: var(--pf-global--Color--dark-200);
   --pf-c-dropdown__menu-item--hover--BackgroundColor: var(--pf-global--BackgroundColor--light-300);
   --pf-c-dropdown__menu-item--disabled--BackgroundColor: transparent;
+  --pf-c-dropdown__menu-item--m-text--Color: var(--pf-global--Color--400);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss]: @mcarrano @maryshak1996 can we verify this is the correct color? This is the only place where we use `--pf-global--Color--400`. In the radio/check description, empty state body, and table caption, we use `--pf-global--Color--200`. Should we be using that instead?

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 5]
```diff
@@ -389,6 +411,17 @@
...
   }
+
+  &.pf-m-text {
+    color: var(--pf-c-dropdown__menu-item--m-text--Color);
+
+    &:hover,
+    &.pf-m-hover,
+    &:focus,
+    &.pf-m-focus {
+      background-color: transparent;
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss]:
```suggestion
      --pf-c-dropdown__menu-item--BackgroundColor: transparent;
```

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 6]
```diff
@@ -389,6 +411,17 @@
     display: flex;
     align-items: center;
   }
+
+  &.pf-m-text {
+    color: var(--pf-c-dropdown__menu-item--m-text--Color);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss]:
```suggestion
    --pf-c-dropdown__menu-item--Color: var(--pf-c-dropdown__menu-item--m-text--Color);
```

File: src/patternfly/components/Page/page.scss [hunk 7]
```diff
@@ -206,6 +212,29 @@
...
     grid-template-columns: auto 1fr auto;
   }
+
+  .pf-c-dropdown {
+    .pf-c-avatar {
+      --pf-c-page__header-tools--c-avatar--MarginLeft: 0;
+    }
+  }
+
+  .pf-c-dropdown__toggle {
```
Comment (mcoker) [inline src/patternfly/components/Page/page.scss]: I wonder if we should just make the dropdown enhancement, then the page enhancements once we finish the overall masthead story? Some questions:

* If someone is using another dropdown in the header, do we want the bottom border styles applied?
* Would we want the underline styles to apply to the existing user dropdown without the avatar?
* Can we handle the bottom border better so that the sides aren't angled?
* Since we can include this in the breaking release, do we want to support using the avatar outside of the dropdown menu anymore? 

Here's a screenshot of another dropdown in the header with the underline styles, also showing the angled sides.

Screen Shot 2020-03-10 at 5 50 09 PM

File: src/patternfly/demos/Page/page-template-header-tools-elements.hbs [hunk 8]
```diff
@@ -9,16 +9,21 @@
...
   {{/page-header-tools-group}}
-  {{#> page-header-tools-group}}
-    {{#> dropdown id=(concat page--id '-dropdown-kebab-right-aligned-1') dropdown--IsActionMenu="true" dropdown-menu--modifier="pf-m-align-right" dropdown-toggle--modifier="pf-m-mobile pf-m-plain" dropdown--HasKebabIcon="true" aria-label="Actions"}}
+  {{#> page-header-tools-group page-header-tools-group--modifier="pf-m-user-dropdown"}}
+    {{#> dropdown
+      id=(concat page--id '-dropdown')
+      dropdown--IsMenuToggleImageText="true"
+      dropdown-menu--modifier="pf-m-align-right"
+      dropdown-menu--dt-modifier="pf-c-page__header-label"
+      dropdown-menu--dd-modifier="pf-c-page__header-text"
```
Comment (mcoker) [inline src/patternfly/demos/Page/page-template-header-tools-elements.hbs]: `dropdown-menu--dt/dd...` don't look to be supported.

File: src/patternfly/demos/Page/page-template-header-tools-elements.hbs [hunk 9]
```diff
@@ -9,16 +9,21 @@
         <i class="fas fa-cog" aria-hidden="true"></i>
     {{/button}}
   {{/page-header-tools-group}}
-  {{#> page-header-tools-group}}
-    {{#> dropdown id=(concat page--id '-dropdown-kebab-right-aligned-1') dropdown--IsActionMenu="true" dropdown-menu--modifier="pf-m-align-right" dropdown-toggle--modifier="pf-m-mobile pf-m-plain" dropdown--HasKebabIcon="true" aria-label="Actions"}}
+  {{#> page-header-tools-group page-header-tools-group--modifier="pf-m-user-dropdown"}}
```
Comment (mcoker) [inline src/patternfly/demos/Page/page-template-header-tools-elements.hbs]: did you mean to add `pf-m-user-dropdown` in this PR? If so you can use that to target the avatar and potentially toggle underline styles if you wan to make all of those changes part of this PR.

File: src/patternfly/components/Dropdown/dropdown.scss (hunk 4 above)
Comment (mattnolting) [reply src/patternfly/components/Dropdown/dropdown.scss]: @mceledonia and I agreed that `--pf-global--Color--200` didn't provide enough contrast. 

@mcarrano @maryshak1996 wdyt?

File: src/patternfly/components/Page/page.scss (hunk 7 above)
Comment (mattnolting) [reply src/patternfly/components/Page/page.scss]: Agree. #2808

File: src/patternfly/components/Dropdown/dropdown-separator.hbs [hunk 10]
```diff
@@ -1,7 +1,8 @@
 <{{#if dropdown-separator--IsHr}}hr{{else}}li{{/if}} class="pf-c-dropdown__separator{{#if dropdown-separator--modifier}} {{dropdown-separator--modifier}}{{/if}}"
   {{#if dropdown-separator--attribute}}
-	  {{{dropdown-separator--attribute}}}
+    {{{dropdown-separator--attribute}}}
   {{/if}}
+  aria-hidden="true"
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown-separator.hbs]: why is this needed?

File: src/patternfly/components/Page/examples/Page.md [hunk 11]
```diff
@@ -156,6 +156,7 @@ This component provides the basic chrome for a page, including sidebar, header,
 | `.pf-c-page__header-nav` | `<div>` |   Creates a container to nest the navigation component in the header. |
 | `.pf-c-page__header-tools` | `<div>` |   Creates a container to nest the icons and menus in header. |
 | `.pf-c-page__header-tools-group` | `<div>` |  Creates a container for grouping sets of icons and menus in header. |
+| `.pf-c-page__header-label` | `<div>` |  Creates a header label. |
```
Comment (mcoker) [inline src/patternfly/components/Page/examples/Page.md]: need to remove this. You could just revert this file.

File: src/patternfly/components/Page/page.scss (hunk 7 above)
Comment (mcarrano) [reply src/patternfly/components/Page/page.scss]: These are all good questions @mcoker . I know that we talked a bit about this last week. I think we were leaning towards making all dropdowns in the masthead have the same appearance, but that would still be unique to the masthead. Do you recall specifics @mceledonia ?

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 12]
```diff
@@ -331,6 +338,21 @@
   }
 }
 
+// Toggle image
+.pf-c-dropdown__toggle-image {
+  display: inline-flex;
+  margin-top: var(--pf-c-dropdown__toggle-image--MarginTop);
+  margin-bottom: var(--pf-c-dropdown__toggle-image--MarginBottom);
+
+  &:not(:first-child) {
```
Comment (christiemolloy) [inline src/patternfly/components/Dropdown/dropdown.scss]: could you add margin-left to above, and then in the selector `&:first-child` remove the margin left?

File: src/patternfly/components/Dropdown/examples/Dropdown.md [hunk 13]
```diff
@@ -216,6 +234,7 @@ The dropdown menu can contain either links or buttons, depending on the expected
 | `.pf-m-action` | `.pf-c-dropdown__toggle.pf-m-split-button` | Modifies the dropdown toggle for when an action is placed beside a toggle button in a split button dropdown. |
 | `.pf-m-hover` | `.pf-c-dropdown__menu-item`, `.pf-c-dropdown__toggle` | Forces display of the hover state of the element. This state is primarily for demonstration purposes and would not normally be used in lieu of the `:hover` pseudo-class. |
 | `.pf-m-focus` | `.pf-c-dropdown__menu-item`, `.pf-c-dropdown__toggle` | Forces display of the focus state of the element. This state is primarily for demonstration purposes and would not normally be used in lieu of the `:focus` pseudo-class. |
+| `.pf-m-text` | `.pf-c-dropdown__menu-item` | Defines a menu item that is non-interactive text. |
```
Comment (christiemolloy) [inline src/patternfly/components/Dropdown/examples/Dropdown.md]:
```suggestion
| `.pf-m-text` | `.pf-c-dropdown__menu-item` | Modifies a menu item to be non-interactive text. |
```

File: src/patternfly/components/Dropdown/dropdown.scss (hunk 12 above)
Comment (mattnolting) [reply src/patternfly/components/Dropdown/dropdown.scss]: After looking at the placement, if `.pf-c-dropdown__toggle-image` precedes `.pf-c-dropdown__toggle-text`, `margin-right` works well. If if follows `.pf-c-dropdown__toggle-icon`, `.pf-c-dropdown__toggle-image` doesn't need `margin-left`.
